index.js

```diff
@@ -826,6 +888,18 @@
     return function(x) { return f(chain(x))(x); };
   }
 
+  //  Function$prototype$alt :: (x -> a) ~> (x -> a) -> x -> a
+  function Function$prototype$alt(f) {
+    var alt_ = this;
+    return function(x) { return alt(alt_(x), f(x)); };
+  }
+
+  //  Function$prototype$concat :: (x -> a) ~> (x -> a) -> x -> a
+  function Function$prototype$concat(f) {
+    var semigroup = this;
+    return function(x) { return concat(semigroup(x), f(x)); };
+  }
+
```

Comment:
Please add tests for `Function$prototype$alt` and `Function$prototype$concat`.

In both cases I'm interested to know whether the order of arguments (to `alt` and `concat` respectively) is forced upon us by the laws, or whether both orders are lawful.